### PR TITLE
Upgrade pravega client

### PR DESCRIPTION
**Pull request by vangork**
Upgrade to the latest commit.
```
Running unittests (/home/luis/projects/nautilus-gstreamer/gstreamer-pravega/target/release/deps/longevity_test-90e180e2f564f685)
{ "type": "suite", "event": "started", "test_count": 0 }
{ "type": "suite", "event": "ok", "passed": 0, "failed": 0, "allowed_fail": 0, "ignored": 0, "measured": 0, "filtered_out": 0, "exec_time": 0.000037991 }
   Doc-tests integration-test
{ "type": "suite", "event": "started", "test_count": 0 }
{ "type": "suite", "event": "ok", "passed": 0, "failed": 0, "allowed_fail": 0, "ignored": 0, "measured": 0, "filtered_out": 0, "exec_time": 0.000048282 }
+ TEST_RESULT=0
+ [[ 0 != \0 ]]
+ exit 0
```

### CODE REVIEW

File: integration-test/src/extreme_tests.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//
+ //
```
Comment (claudiofahey) [inline integration-test/src/extreme_tests.rs:1]: Remove extra space.

**Review by claudiofahey — CHANGES_REQUESTED** [review]
Looks great! Just 1 small formatting change.

File: integration-test/src/extreme_tests.rs (hunk 1 above)
Comment (claudiofahey) [reply integration-test/src/extreme_tests.rs]: Fixed.

File: gst-plugin-pravega/src/pravegasrc/imp.rs [hunk 2]
```diff
@@ -161,8 +161,9 @@ impl Default for Settings {
 enum State {
     Stopped,
     Started {
-        reader: Arc<Mutex<CountingReader<BufReader<SeekableTake<ByteReader>>>>>,
-        index_searcher: Arc<Mutex<IndexSearcher<ByteReader>>>,
+        reader: Arc<Mutex<CountingReader<BufReader<SeekableTake<SyncByteReader>>>>>,
+        index_searcher: Arc<Mutex<IndexSearcher<SyncByteReader>>>,
+        client_factory: ClientFactory,
```
Comment (claudiofahey) [inline gst-plugin-pravega/src/pravegasrc/imp.rs:166]: Compile shows: 
```
warning: field is never read: `client_factory`
```

**Review by claudiofahey — CHANGES_REQUESTED** [review]
Please fix the compile warning.

File: gst-plugin-pravega/src/pravegasrc/imp.rs (hunk 2 above)
Comment (vangork) [reply gst-plugin-pravega/src/pravegasrc/imp.rs:167]: add the comment at line 166. client_factory has to be saved to keep the tokio runtime.

**Review by claudiofahey — APPROVED** [review]

**Pull request merged**